docs/topics/migrations.txt

```diff
@@ -77,16 +77,13 @@ meaning that if a migration fails to apply you will have to manually unpick
 the changes in order to try again (it's impossible to roll back to an
 earlier point).
 
-In addition, MySQL will fully rewrite tables for almost every schema operation
-and generally takes a time proportional to the number of rows in the table to
-add or remove columns. On slower hardware this can be worse than a minute per
-million rows - adding a few columns to a table with just a few million rows
-could lock your site up for over ten minutes.
-
-Finally, MySQL has relatively small limits on name lengths for columns, tables
-and indexes, as well as a limit on the combined size of all columns an index
-covers. This means that indexes that are possible on other backends will
-fail to be created under MySQL.
```

Comment:
The paragraph I removed talks about name length limits which are the same as postgres (as I pointed out in the PR description).

I've looked at the link you shared, although these limitations are good to know, I don't see how it limits Django (as compared to other backends which have their own limits too).

If you'd like, let me know which limitations you think are important and I'll word them in this PR!

Reply:
The most important ones that come to mind is

> The index key prefix length limit is 767 bytes for InnoDB tables that use the [REDUNDANT](https://dev.mysql.com/doc/refman/8.0/en/glossary.html#glos_redundant_row_format) or [COMPACT](https://dev.mysql.com/doc/refman/8.0/en/glossary.html#glos_compact_row_format) row format. For example, you might hit this limit with a [column prefix](https://dev.mysql.com/doc/refman/8.0/en/glossary.html#glos_column_prefix) index of more than 191 characters on a TEXT or VARCHAR column, assuming a utf8mb4 character set and the maximum of 4 bytes for each character.

This means the index created on a `CharField(max_length>191)` won't be effective and that unique constraint on such fields will only be for the first characters. See ticket-31144 and [this part of the documentation](https://docs.djangoproject.com/en/5.0/ref/databases/#mysql-character-fields)

As for the index name length see ticket-12977.

Reply:
I agree with Simon we should keep the last paragraph. Please restore it.

Reply:
@betaflag Do you have time to keep working on this? I'd leave information about the index limitation:
```diff
diff --git a/docs/topics/migrations.txt b/docs/topics/migrations.txt
index 248a7addbe..bd7b08b34e 100644
--- a/docs/topics/migrations.txt
+++ b/docs/topics/migrations.txt
@@ -83,10 +83,9 @@ add or remove columns. On slower hardware this can be worse than a minute per
 million rows - adding a few columns to a table with just a few million rows
 could lock your site up for over ten minutes.
 
-Finally, MySQL has relatively small limits on name lengths for columns, tables
-and indexes, as well as a limit on the combined size of all columns an index
-covers. This means that indexes that are possible on other backends will
-fail to be created under MySQL.
+Finally, MySQL has a relatively small limit on the combined size of all columns
+an index covers. This means that indexes that are possible on other backends
+will fail to be created under MySQL.
 
 SQLite
 ------

```

Reply:
Thanks for the wonderful feedback, I'll update it on Monday :)

Reply:
@betaflag I can push final [edits](https://github.com/django/django/pull/17582#discussion_r1420381050) if you don't have time to working on this.

Reply:
@felixxm I've pushed the edits, let me know if this is works for you!